### PR TITLE
API: `Styler.hide_columns` replaced by `.hide_values` which also operates row-wise

### PR DESCRIPTION
**Pull request by attack68**
For showing and hiding columns/rows `Styler` currently has two methods:

 - `.hide_columns(subset)`: which hides a selection of columns and the associated data values.
 - `.hide_index()`: which hides the index keys completely and displays all the data values.

This PR adds a `.hide_values(subset, axis, show)` method, which is a superset of `.hide_columns`. It allows the same functionality to operate row-wise, and adds the kwarg `show` which allows an inverse method, exclusively showing instead of hiding.

#### Option 1

Keep `.hide_values(subset, axis, show)` as is and **deprecate** `.hide_columns`.

#### Option 2

Make `._hide_values(subset, axis, show)` a private module and include/add the public methods:

    - `hide_columns`      = `_hide_values(subset=subset, axis="columns", show=False)`.
    - `hide_rows`         = `_hide_values(subset=subset, axis="index", show=False)`.
    - `show_columns`      = `_hide_values(subset=subset, axis="columns", show=True)`.
    - `show_rows`         = `_hide_values(subset=subset, axis="index", show=True)`.

### CODE REVIEW

File: pandas/io/formats/style.py [hunk 1]
```diff
@@ -1060,12 +1061,76 @@ def hide_columns(self, subset) -> Styler:
...
         """
-        subset = non_reducing_slice(subset)
-        hidden_df = self.data.loc[subset]
-        hcols = self.columns.get_indexer_for(hidden_df.columns)
-        # error: Incompatible types in assignment (expression has type
-        # "ndarray", variable has type "Sequence[int]")
-        self.hidden_columns = hcols  # type: ignore[assignment]
+        return self.hide_values(subset)
+
+    def hide_values(self, subset, axis: Axis = "columns", show: bool = False) -> Styler:
```
Comment (jreback) [inline pandas/io/formats/style.py:1066]: i find this a very confusing name, hide_axis is more appropriate

File: pandas/io/formats/style.py [hunk 2]
```diff
@@ -1060,12 +1061,76 @@ def hide_columns(self, subset) -> Styler:
...
+        """
```
Comment (jreback) [inline pandas/io/formats/style.py:1067]: i wouldn't make this user visible instead prefer hide_index, hide_columns (this can be the impl)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/formats/style.py (hunk 1 above)
Comment (attack68) [reply pandas/io/formats/style.py:1066]: The trick is to distinguish between the cases of:

 a) hiding select rows or columns of data (whilst other index keys or column headers are visible),
 b) or displaying the data values but just hiding the index or column headers row in their entirety.

Currently `hide_columns` does a) whilst `hide_index` does b). The complete idea was that:

 i) `hide_values` would do a) for either the index or columns axes.
ii) `hide_headers` (a new method) would do b) for either the index or columns axes.

?

Comment (attack68) [reply pandas/io/formats/style.py:1066]: @jreback i created an alternative in #41266 which seeks to achieve the same functionality in a different way re-using exiting methods and kwargs. Close this PR if you prefer the alternative,

**Comment by attack68** [comment]
closing this is favour of #41266 for now.

**Pull request closed without merging**